app/models/locations/map.rb

```diff
@@ -0,0 +1,99 @@
+module Locations
+  class Map
+    attr_accessor :id,:loc
+
+    def initialize(id: nil, loc: nil)
+      @locator_url = "http://library.princeton.edu/locator/index.php?"
+      @stackmap_url = "http://princeton.stackmap.com/view/?"
+      @id ||= id
+      @loc ||= loc
+    end
+
+    def url
+      if is_valid?
+        if self.locator_libs.include? self.lib.code
```

Comment:
I remember when discussed this last @tpendragon made a good suggestion about making nested conditionals more readable with guard statements http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html. Might help make this longer method a bit more readable.